fv-all-in-one-seo-pack.php

```diff
@@ -3688,6 +3696,102 @@ function script_footer_content(){
 
 
 
+  function manage_category_columns( $columns ){
+    add_action('admin_footer', array($this,'manage_category_fvseo_title_js') );
+    
+    $new_columns  = array_slice($columns, 0, 2)
+                  + array('fvseo_title' => "SEO Title")
+                  + array_slice($columns, 2);
+    
+    return $new_columns;
+  }
+
+
+
+
+  function manage_category_custom_columns( $content, $column_name, $term_id ){
+    if( $column_name != 'fvseo_title' ){
+      return $content;
+    }
+    
+    $category_titles = get_option('aioseop_category_titles');
+    $value = ( isset($category_titles[$term_id]) && strlen(trim($category_titles[$term_id])) > 0 ) ? $category_titles[$term_id] : '';
+    
+    $content .= "<input class='fvseo_title' type='text' name='fvseo_title[$term_id]' value='$value'>";
+
+    return $content;
+  }
+  
+  
+  
+  
+  function manage_category_process_action(){
+    if( !isset( $_POST['fv_seo_category_update'] ) ){
+      return;
+    }
+    
+    $seo_titles = $_POST['fvseo_title'];
+    if( isset($seo_titles) && !empty($seo_titles) ){
+      $category_titles = get_option('aioseop_category_titles');
+      
+      if( !$category_titles){
+        $category_titles = array();
+      }
+      
+      foreach($seo_titles as $term_id => $title ){
+        if(  strlen(trim($title)) > 0 ){
+          $category_titles[$term_id] = $title;
+        }
+      }
+      
+      update_option('aioseop_category_titles',$category_titles);
+    }
+    
+    //clear after process, 
+    $_POST = array();
+  }
+  
+  
+  function manage_category_fvseo_title_js(){
+  ?>
```

Comment:
I would go for separation of JS into separate file and not inlining it. This file already has 3400 more lines it should have ;-)